nav2_controller/src/nav2_controller.cpp

```diff
@@ -68,7 +68,13 @@ ControllerServer::ControllerServer()
 ControllerServer::~ControllerServer()
 {
   progress_checker_.reset();
-  goal_checker_.reset();
+
+  for (auto & goal_checker: goal_checkers_) {
+    goal_checker.second.reset();
```

Comment:
This will be done automatically in L76 when you reset the entire vector container, the destructors for each unique pointer will be called

Reply:
Okay, I removed that part.
I also removed the part of the goal_checker_ids.clear() to totally mimic the usage of the controllers vector (we did not have controller_ids.clear())